hibernate-core/src/test/java/org/hibernate/event/PreUpdateEventListenerVetoTest.java

```diff
@@ -0,0 +1,76 @@
+package org.hibernate.event;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Version;
+
+import org.hibernate.event.service.spi.EventListenerRegistry;
+import org.hibernate.event.spi.EventType;
+import org.hibernate.event.spi.PreUpdateEventListener;
+
+import org.hibernate.testing.TestForIssue;
+import org.hibernate.testing.junit4.BaseCoreFunctionalTestCase;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.hibernate.testing.transaction.TransactionUtil.doInHibernate;
+
+/**
+ * @author Nathan Xu
+ */
+@TestForIssue( jiraKey = "HHH-14413" )
+public class PreUpdateEventListenerVetoTest extends BaseCoreFunctionalTestCase {
+
+	private static final Long EXAMPLE_ID_VALUE = 1L;
+
+	@Override
+	protected Class<?>[] getAnnotatedClasses() {
+		return new Class<?>[] { ExampleEntity.class };
+	}
+
+	@Override
+	protected void afterSessionFactoryBuilt() {
+		super.afterSessionFactoryBuilt();
+		EventListenerRegistry registry = sessionFactory().getServiceRegistry().getService( EventListenerRegistry.class );
+		registry.appendListeners(
+				EventType.PRE_UPDATE,
+				(PreUpdateEventListener) event -> event.getEntity() instanceof ExampleEntity
+		);
+	}
+
+	@Before
+	public void setUp() {
+		doInHibernate( this::sessionFactory, session -> {
+			ExampleEntity entity = new ExampleEntity();
+			entity.id = EXAMPLE_ID_VALUE;
+			entity.name = "old_name";
+			session.save( entity );
+		} );
+	}
+
+	@Test
+	public void testVersionNotTouchedWhenPreUpdateEventVetoed() {
+		doInHibernate( this::sessionFactory, session -> {
+			ExampleEntity entity = session.byId( ExampleEntity.class ).load( EXAMPLE_ID_VALUE );
+			entity.name = "new_name";
+			session.update( entity );
+
+			// without fixing HHH-14413, the following statement would fail due to incremented version
+			// even though updating has been vetoed
+			session.delete( entity );
```

Comment:
Probably we should test the version directly as well, because the delete statement´s inner working could be changed. I propose additionally to the current test:
```
ExampleEntity entity = session.byId( ExampleEntity.class ).load( EXAMPLE_ID_VALUE );
Long initialVersion = entity.version;
//...
session.update( entity );
assertEquals(initialVersion, entity.version, "The entity version must not change when update is vetoed");
//...
```

Reply:
Good idea. It also align well with the testing method name. 
The issue is `entity.version` doesn't change. What is changed is the internal `EntityPersister`. I am afraid we have to go about low level testing which is tightly coupled with implementation detail.
The current approach is the only means I could think of without testing internal implementation detail. Let me know your thoughts.

Reply:
Finally I figured it out. We need to invoke `session.flush()` before assertion. Then it works perfectly! Thanks again for your good bug reporting and code review.

Reply:
Not sure if I misunderstand these lines in `postUpdate(...)`:
```java
		if ( getPersister().isVersioned() ) {
			this.version = nextVersion;
			getPersister().setPropertyValue( entity, getPersister().getVersionProperty(), nextVersion );
		}
```
My understanding is that this does not change the persister, but the entity. Please correct me if I am wrong. So the value of the version property should actually be changed.

Reply:
Alright, during me writing this you already found the solution :D Thank you!

Reply:
btw, I also included your name as the second author of the testing case, :).

Reply:
That´s great to hear, thanks for doing the heavy lifting of implementation, I would not have been able to make the test case that concise.